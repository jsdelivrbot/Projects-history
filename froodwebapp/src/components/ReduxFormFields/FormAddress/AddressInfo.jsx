import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'components';
import { Icon } from 'antd';
import {
  addressInfo,
  addressUserName,
  editIcon
} from './AddressInfo.scss';

class AddressInfo extends Component {
  handleAddressClick = () => {
    if (this.props.addressClick) {
      this.props.addressClick(this.props.user.address.id);
    }
  }

  render() {
    const {
      user,
      editIconVisible = false,
      editIconClick
    } = this.props;

    return (
      <Button
        className={ addressInfo }
        onClick={ this.handleAddressClick }
      >
        <div className={ addressUserName }>
          <strong>{ user.name }</strong>
          { editIconVisible &&
            <Icon
              id={ user.address.id }
              className={ editIcon }
              type="edit"
              onClick={ editIconClick }
            />
          }
        </div>
        { user.address &&
          <span>
            { user.address.address1 && <div>{ user.address.address1 }</div> }
            { user.address.address2 && <div>{ user.address.address2 }</div> }
            { user.address.cityName && <div>{ `${user.address.cityName} ${user.address.postalCode || ''}` }</div> }
            { user.address.countryName && <div>{ user.address.countryName }</div> }
            { user.address.countryCode &&
              <div id="phoneNo">
                { `${user.address.countryCode || ''}${user.address.phone || ''}` }
              </div>
            }
          </span>
        }
        { !user.address &&
          <span>
            <div>There is no address</div>
            <div>Please, add new one</div>
          </span>
        }
      </Button>
    );
  }
}

AddressInfo.propTypes = {
  user: PropTypes.object,
  addressClick: PropTypes.func,
  editIconVisible: PropTypes.bool,
  editIconClick: PropTypes.func,
};

export default AddressInfo;
