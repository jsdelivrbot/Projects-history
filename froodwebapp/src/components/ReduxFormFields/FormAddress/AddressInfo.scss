@import '~styles/colors.scss';

$main-icon-size: 1.2rem;
$main-font-size: 14px;
$company-font-size: $main-font-size + 2px;

.addressUserName {
  display: flex;
  align-content: space-between;
  align-items: center;
  strong {
   flex: 1
  }
  font-size: $company-font-size;
}

.addressInfo {
  div {
    line-height: 1.5rem;
  }
  &:hover {
    color: $main-black;
  }
  width: 100%;
  padding: 0;
  color: $main-black;
  font-size: $main-font-size;
  border: none;
  background: transparent;
  height: auto;
  text-align: left;
  div#phoneNo {
    font-size: $main-font-size;
    margin-bottom: 0;
  }
}

.editIcon {
  margin-left: 1rem;
  font-size: $main-icon-size;
  color: $main-black;
  cursor: pointer;
}