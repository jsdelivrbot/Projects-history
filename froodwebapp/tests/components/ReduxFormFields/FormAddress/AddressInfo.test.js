import React from 'react';
import AddressInfo from 'components/ReduxFormFields/FormAddress/AddressInfo';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const { describe, it, expect } = global;

describe('AddressInfo', () => {
  it('captures snapshot of AddressInfo', () => {
    const props = {
      user: {
        address: {
          address1: 'address1',
          address2: 'address2',
          cityName: 'LA',
          countryName: 'USA',
          id: '1'
        },
        name: ''
      },
      addressClick: jest.fn(),
      editIconVisible: false,
      editIconClick: jest.fn()
    };
    const snappedComponent = shallow(<AddressInfo { ...props } />);

    expect(toJson(snappedComponent)).toMatchSnapshot();
  });
});
