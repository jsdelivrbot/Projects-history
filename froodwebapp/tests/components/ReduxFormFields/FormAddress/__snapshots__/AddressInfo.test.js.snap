// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressInfo captures snapshot of AddressInfo 1`] = `
<Button
  className="addressInfo"
  onClick={[Function]}
>
  <div
    className="addressUserName"
  >
    <strong />
  </div>
  <div>
    address1
  </div>
  <div>
    address2
  </div>
  <div>
    LA 
  </div>
  <div>
    USA
  </div>
</Button>
`;
